addon/components/x-option.js

```diff
@@ -32,15 +32,22 @@ export default Ember.Component.extend({
    * `<option>` element. It is aware of the containing `x-select`'s
    * value and will mark itself if it is the same.
    *
+   * If a PromiseObject is passed as the selection value,
+   * the `content` property of that PromiseObject will be evaluated.
+   *
    * @private
    * @property selected
    * @type Boolean
    */
   selected: Ember.computed('value', 'select.value', 'select.multiple', function() {
-    if (this.get('select.multiple') && isArray(this.get('select.value'))) {
-      return this.get('select.value').contains(this.get('value'));
+    var selectValue = this.get('select.value');
+    if(selectValue && selectValue.content) {
```

Comment:
http://emberjs.com/api/data/classes/DS.PromiseObject.html set's the `content` property once the Promise is resolved. Ideally it should be selectValue.then( fn(content) { ... }), but as far as I know ComputedProperties don't do async.